timesketch/api/v1/resources/archive.py

```diff
@@ -343,13 +343,13 @@ def _export_view(self, view, sketch, zip_file):
         """
         name = "{0:04d}_{1:s}".format(view.id, view.name)
 
-        query_filter = json.loads(view.query_filter)
+        if view.query_filter:
+            query_filter = json.loads(view.query_filter)
+
         if not query_filter:
             query_filter = self.DEFAULT_QUERY_FILTER
 
-        indices = query_filter.get("indices", self.sketch_indices)
-        if not indices or "_all" in indices:
-            indices = self.sketch_indices
+        indices = self.sketch_indices
```

Comment:
I know that this is already merged, but this block means that the saved search will be wrong in the case that the filter has specific indices selected, and will potentially return unexpected result, which is probably not what we want?

Reply:
ping @jaegeral who reviewed this PR

Reply:
Ya, i mentioned this in the comment above. 

"Not sure if this is the best way to do it, and also could find the origin of the wrong index number in the query_filter. hence this is just a Proposal, and will be waiting for input regarding this."

Reply:
There is something wrong with 

```
query_filter = json.loads(view.query_filter)
indices = query_filter.get("indices", self.sketch_indices)
```

Not sure where is the index in the query filter is set. it returned the wrong index. It was just a 1 digit int